cvmfs/file_watcher_kqueue.cc

```diff
@@ -0,0 +1,164 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <fcntl.h>
+#include <sys/event.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <string>
+#include <vector>
+
+#include "backoff.h"
+#include "file_watcher_kqueue.h"
+#include "logging.h"
+#include "util/posix.h"
+
+namespace file_watcher {
+
+FileWatcherKqueue::FileWatcherKqueue() {}
+
+FileWatcherKqueue::~FileWatcherKqueue() {}
+
+bool FileWatcherKqueue::RunEventLoop(const FileWatcher::HandlerMap& handlers,
+                                     int control_pipe) {
+  kq_ = kqueue();
+  if (kq_ == -1) {
+    LogCvmfs(kLogCvmfs, kLogDebug, "FileWatcherKqueue - Cannot create kqueue.");
+    return false;
+  }
+
+  // Control pipe sending the stop event.
+  struct kevent watch_event;
+  EV_SET(&watch_event, control_pipe, EVFILT_READ, EV_ADD | EV_ENABLE | EV_CLEAR,
```

Comment:
I guess we can assert on a positive result here, too.

Reply:
I think EV_SET returns void, not sure what to assert.